bodo/io/parquet_pio.py

```diff
@@ -657,6 +657,39 @@ def unify_schemas_across_ranks(dataset: ParquetDataset, total_rows_chunk: int):
     ev.finalize()
 
 
+def unify_fragment_schema(dataset: ParquetDataset, piece: ParquetPiece, frag):
+    """Unifies schema of *dataset* with incoming piece/fragment.
+
+    Args:
+        dataset (ParquetDataset): The Parquet dataset with schema to unify.
```

Comment:
Maybe some indication here that it is this argument that is updated with the unified schema.